dependency/vault_pki.go

```diff
@@ -127,9 +127,12 @@ func goodFor(cert *x509.Certificate) (time.Duration, bool) {
 	lifespanDur := end.Sub(start)
 	r := rand.New(rand.NewSource(time.Now().UnixNano()))
 	lifespanMilliseconds := lifespanDur.Milliseconds()
-	// calculate the 'time the certificate should be rotated' by figuring out
-	// 87-93% of the lifespan and adding it to the start
-	rotationTime := start.Add(time.Millisecond * time.Duration(((lifespanMilliseconds*9)/10)+(lifespanMilliseconds*int64(r.Intn(6)-3))/100))
+	// calculate the 'time the certificate should be rotated' by figuring out -2%
+	// - 3% + VaultLeaseRenewalThreshold of the lifespan and adding it to the
+	// start
+	rotationTime := start.Add(time.Millisecond * time.Duration(
+		float64(lifespanMilliseconds)*VaultLeaseRenewalThreshold+float64(lifespanMilliseconds*(int64(r.Intn(6)-3)/100.0)),
+	))
```

Comment:
There's a linter failure having to do with conversions that might be worth checking. There is also another linter failure on line [dependency/vault_pki.go#L136](https://github.com/hashicorp/consul-template/pull/1872/files#annotation_20734928620) ] that we forgot to correct in the last PR. If we could fix that, that would be amazing!